### PR TITLE
skip reordering elements with no parent, fix phx-remove on re-added stream items

### PR DESCRIPTION
**Pull request by SteffenDE**
This would throw otherwise while trying to reorder an element that is not in the DOM.

We also had a bug where stream items with phx-remove were removed, despite re-adding them in a stream reset.

### CODE REVIEW

**Comment by DaTrader** [comment]
@chrismccord Can you please merge this one too? It's the hopefully final bug fix for stream reset. 

Thanks

**Comment by chrismccord** [comment]
If it’s here it’s on my list already

File: assets/js/phoenix_live_view/dom_patch.js [hunk 1]
```diff
@@ -365,6 +366,12 @@ export default class DOMPatch {
       return
     }
 
+    // check if the element has a parent element;
+    // it doesn't if we are currently recursively morphing (restoring a saved stream child)
+    // because the element is not yet added to the real dom;
+    // reordering does not make sense in that case anyway
+    if(!el.parentElement) return
```
Comment (chrismccord) [inline assets/js/phoenix_live_view/dom_patch.js:373]:
```suggestion
    if(!el.parentElement){ return }
```

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**